securecookie.go

```diff
@@ -325,6 +328,7 @@ func (s *SecureCookie) Decode(name, value string, dst interface{}) error {
 	if t1, err = strconv.ParseInt(string(parts[0]), 10, 64); err != nil {
 		return errTimestampInvalid
 	}
+	s.LastTimestamp = t1
```

Comment:
The race condition definitely exists. This whole concept depends on using a single instance of `SecureCookie` in only one thread. It's just that with using the local variable, users who don't care about `LastTimestamp` aren't affected in any way.